sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -2437,6 +2439,30 @@ def wait_for_port_config_done(self, namespace):
             if key in ["PortConfigDone", "PortInitDone"]:
                 break
 
+    """
+    Initialize NPU_SI_SETTINGS_SYNC_STATUS_KEY field in STATE_DB PORT_TABLE|<lport>
```

Comment:
@mihirpat1 where do we delete the key when optics is removed?

Reply:
@prgeor We don't delete the key when optics is removed. We instead mark it as `NPU_SI_SETTINGS_DEFAULT`.
This change is at line 2163-2164 in xcvrd.py.